lib/ansible/modules/database/postgresql/postgresql_facts.py

```diff
@@ -0,0 +1,916 @@
+            'disable', 'allow', 'prefer', 'require', 'verify-ca', 'verify-full']),
+        ssl_rootcert=dict(default=None, type=str),
+    ))
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True
+    )
+
+    if not HAS_PSYCOPG2:
+        module.fail_json(msg="The python psycopg2 module is required")
+
+    incl_subsets = module.params["incl_subset"]
+    excl_subsets = module.params["excl_subset"]
+    sslrootcert = module.params["ssl_rootcert"]
+
+    if incl_subsets != "*" and excl_subsets:
+        module.fail_json(msg="incl_subset and excl_subset parameters "
+                             "are mutually exclusive")
+
+    # To use defaults values, keyword arguments must be absent, so
+    # check which values are empty and don't include in the **kw
+    # dictionary
+    params_map = {
+        "login_host": "host",
+        "login_user": "user",
+        "login_password": "password",
+        "port": "port",
+        "db": "database",
+        "ssl_mode": "sslmode",
+        "ssl_rootcert": "sslrootcert"
+    }
+    kw = dict((params_map[k], v) for (k, v) in iteritems(module.params)
+              if k in params_map and v != "" and v is not None)
+
+    # If a login_unix_socket is specified, incorporate it here.
+    is_localhost = "host" not in kw or kw["host"] == "" or kw["host"] == "localhost"
+    if is_localhost and module.params["login_unix_socket"] != "":
+        kw["host"] = module.params["login_unix_socket"]
+
+    if psycopg2.__version__ < '2.4.3' and sslrootcert is not None:
+        module.fail_json(msg='psycopg2 must be at least 2.4.3 in order to user the ssl_rootcert parameter')
+
+    try:
+        db_connection = psycopg2.connect(**kw)
+
+        cursor = db_connection.cursor(cursor_factory=psycopg2.extras.DictCursor)
+    except TypeError as e:
+        if 'sslrootcert' in e.args[0]:
+            module.fail_json(msg='Postgresql server must be at least version 8.4 to support sslrootcert')
+        module.fail_json(msg="unable to connect to database: %s" % to_native(e),
+                         exception=traceback.format_exc())
+    except Exception as e:
+        module.fail_json(msg="unable to connect to database: %s" % to_native(e),
+                         exception=traceback.format_exc())
+
+    # Do job:
+    pg_facts = PgClusterFacts(module, cursor)
+
+    if incl_subsets != "*":
+        incl_subsets = [s.strip(" \'][") for s in incl_subsets.split(',')]
+        kw['ansible_facts'] = pg_facts.collect(include=incl_subsets)
+
+    elif excl_subsets:
+        excl_subsets = [s.strip(" \'][") for s in excl_subsets.split(',')]
+        kw['ansible_facts'] = pg_facts.collect(exclude=excl_subsets)
+
+    else:
+        kw['ansible_facts'] = pg_facts.collect()
+
+    # Rollback transaction, if checkmode.
+    # Otherwise, commit transaction:
+    # (It doesn't make sense in this praticular case, just for order)
+    if module.check_mode:
```

Comment:
Please get rid of this whole block. We should be committing or rolling back anything, it is a facts-module.